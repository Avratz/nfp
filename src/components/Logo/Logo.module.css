.logo {
  display: flex;
  align-items: center;
  gap: 0.5rem;
  position: relative;
  z-index: 100;
}

.isotipo {
  width: 32px;
  height: 32px;
  position: relative;
  display: inline-block;
}

.heading {
  letter-spacing: -0.5px;
  font-family: var(--font-poppins) !important;
}

.white {
  fill: currentColor;
  color: var(--neutral-8);
}

.dark {
  fill: currentColor;
  color: var(--neutral-1);
}
