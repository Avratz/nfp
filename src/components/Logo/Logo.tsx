import { cx } from 'class-variance-authority'
import PaisanosFlag from '@/assets/images/paisanos-flag.svg'

import { Link } from '@/ui/Link/Link'
import { Heading } from '@/ui/Heading/Heading'

import styles from './Logo.module.css'

type LogoProps = {
  className?: string
  dark?: boolean
  href?: string
}

export function Logo({ className = '', dark, href = '/' }: LogoProps) {
  return (
    <Link
      href={href}
      className={cx(styles.logo, className, dark ? styles.dark : '')}
    >
      <picture className={styles.isotipo}>
        <PaisanosFlag className={dark ? styles.dark : styles.white} />
      </picture>
      <Heading
        size="2xl"
        inlineBlock
        className={cx(styles.heading, dark ? styles.dark : '')}
      >
        NFPaisanos
      </Heading>
    </Link>
  )
}
