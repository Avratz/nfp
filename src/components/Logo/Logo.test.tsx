import { render } from '@testing-library/react'
import { Logo } from './Logo'

jest.mock('@/assets/images/paisanos-flag.svg', () => {
  const PaisanosFlag = () => null
  PaisanosFlag.displayName = 'PaisanosFlag'
  return PaisanosFlag
})

describe('Logo', () => {
  it('renders Logo', () => {
    let { container } = render(<Logo />)
    expect(container).toMatchSnapshot()
  })
})
