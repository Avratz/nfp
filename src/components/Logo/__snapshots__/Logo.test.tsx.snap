// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders Logo 1`] = `
<div>
  <a
    class="link logo"
    href="/"
  >
    <picture
      class="isotipo"
    />
    <h1
      class="heading 2xl inlineBlock heading"
    >
      NFPaisanos
    </h1>
  </a>
</div>
`;
